signup-service/src/main/java/io/mosip/signup/services/RegistrationService.java

```diff
@@ -166,11 +167,14 @@ public GenerateChallengeResponse generateChallenge(GenerateChallengeRequest gene
 
         HashMap<String, String> hashMap = new LinkedHashMap<>();
         hashMap.put("{challenge}", challenge);
-        notificationHelper.sendSMSNotificationAsync(generateChallengeRequest.getIdentifier(), transaction.getLocale(),
-                        SEND_OTP_SMS_NOTIFICATION_TEMPLATE_KEY, hashMap)
-                .thenAccept(notificationResponseRestResponseWrapper ->
-                    log.debug(notificationLogging, notificationResponseRestResponseWrapper)
-                );
+        RestResponseWrapper<NotificationResponse> notificationResponseRest;
+        try{
+            notificationResponseRest = notificationHelper.sendSMSNotification(generateChallengeRequest.getIdentifier(), transaction.getLocale(),
+                    SEND_OTP_SMS_NOTIFICATION_TEMPLATE_KEY, hashMap);
+            log.debug(notificationLogging, notificationResponseRest);
+        }catch (RestClientException e){
+            log.debug(notificationLogging, "send notification failed");
```

Comment:
If there is a failure to send OTP SMS, then generate-challenge endpoint response should return an error response.